highfidelity/hifi
# 2d images (JPG and PNG) are now supported as (quasi) entities

## Test Plan

1. Open "Create Mode"
2. Click on the "Image" button
3. An image of a dog should appear in front of you
4. Look at the properties of the image, it should be listed as an "Image" and there should be an "Image URL" field
5. Copy a link to a .jpg or .png into that field (e.g. https://static.boredpanda.com/blog/wp-content/uuuploads/cute-baby-animals-2/cute-baby-animals-2-2.jpg)
6. The image model should change to display your new image
7. Drag a .jpg or a .png into Interface
8. A new image entity should appear in front of you with your image
9. Open up the Asset Browser
10. Find the image that you dragged in the list or use a different image in your asset browser
11. Click on "Add To World"
12. That image should appear in front of you

### Side Notes
Functionality is complete, but changes still to be made:

- The "Image Entity Type" is based on existing model entity properties. An unfinished implementation of a separate image entity type is in this PR: https://github.com/highfidelity/hifi/pull/12153
- Use a different model for the model URL other than the snap model
- Need a .svg icon for Images in the Create menu
- Need a default image when no URL has been input
- If possible, auto-resizing the model to fit the image
